### PR TITLE
[miio] add support for cgllc.airm.cgdn1 Qingping Air Monitor Lite

### LINKED ISSUE #10152: [miio] Please support Qingping Air Monitor Lite (cgllc.airm.cgdn1)
@marcelrv Please support Qingping Air Monitor Lite (cgllc.airm.cgdn1)

See Miot specs: https://miot-spec.org/miot-spec-v2/instance?type=urn:miot-spec-v2:device:air-monitor:0000A008:cgllc-cgdn1:1

And the integration in python-miio for reference
https://github.com/rytilahti/python-miio/commit/78f8beee13d9fbd5f2b1a11a311f736b41a2ba06

### PR DESCRIPTION
**Pull request by marcelrv**
close #10152

Signed-off-by: Marcel Verpaalen <marcel@verpaalen.com>

### CODE REVIEW

File: bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json [hunk 1]
```diff
@@ -0,0 +1,492 @@
...
+				"actions": [],
+				"readmeComment": "Value mapping [\"1\"\u003d\"Charging\",\"2\"\u003d\"Not charging\",\"3\"\u003d\"Not chargeable\"]"
+			},
+			{
+				"property": "voltage",
+				"siid": 4,
+				"piid": 3,
+				"friendlyName": "Battery - Voltage",
+				"channel": "voltage",
+				"type": "Number",
```
Comment (cpmeister) [inline bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json:289]:
```suggestion
				"type": "Number:ElectricPotential",
```

File: bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json [hunk 2]
```diff
@@ -0,0 +1,492 @@
...
+				"refresh": true,
+				"actions": []
+			},
+			{
+				"property": "start-time",
+				"siid": 9,
+				"piid": 2,
+				"friendlyName": "Settings - Start Time",
+				"channel": "start_time",
+				"type": "Number",
```
Comment (cpmeister) [inline bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json:319]: Is this supposed to be a time unit?

File: bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json [hunk 3]
```diff
@@ -0,0 +1,492 @@
...
+				"refresh": true,
+				"actions": []
+			},
+			{
+				"property": "pm10-density",
+				"siid": 3,
+				"piid": 5,
+				"friendlyName": "Environment - PM10 Density",
+				"channel": "pm10_density",
+				"type": "Number",
```
Comment (cpmeister) [inline bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json:190]:
```suggestion
				"type": "Number:Density",
```

File: bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json [hunk 4]
```diff
@@ -0,0 +1,492 @@
...
+				"refresh": true,
+				"actions": []
+			},
+			{
+				"property": "pm2.5-density",
+				"siid": 3,
+				"piid": 4,
+				"friendlyName": "Environment - PM2 5 Density",
+				"channel": "pm2_5_density",
+				"type": "Number",
```
Comment (cpmeister) [inline bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json:172]:
```suggestion
				"type": "Number:Density",
```

**Review by cpmeister — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json (hunk 2 above)
Comment (marcelrv) [reply bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json:320]: @joeberlin123 can you confirm what you are seeing for this channel as value and how it translates in the mihome app.
Than we can determine what is indeed it is (e.g. the time until it switches on/off or a date/time sort of thing.

The spec does not hint anything

File: bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json (hunk 1 above)
Comment (marcelrv) [reply bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json]: @joeberlin123 For this, can you confirm what you see as value in the screen of the app and in the channel.
I expect this to be mV, but would be good to confirm

File: bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json (hunk 3 above)
Comment (marcelrv) [reply bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json]: Agree, Done.

File: bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json (hunk 4 above)
Comment (marcelrv) [reply bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json]: Agree, Done.

File: bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json (hunk 1 above)
Comment (joeberlin123) [reply bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json]: Yes, I can confirm. The channel shows the values as mV

File: bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json (hunk 2 above)
Comment (joeberlin123) [reply bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json:320]: I don't find something like start time in the mi home app. If I use item type number I get 0.0 and if I use date time I get NULL. Do have another hint for me?

**Review by fwolter — CHANGES_REQUESTED** [review]
You code doesn't compile as `CloudLoginDTO` uses the wrong `NotNull` import:
`import org.jetbrains.annotations.NotNull;`

File: bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json (hunk 2 above)
Comment (marcelrv) [reply bundles/org.openhab.binding.miio/src/main/resources/database/cgllc.airm.cgdn1-miot.json]: I took out this property... if we can't find it in the app and it is unknown what it does, it is useless.

**Comment by marcelrv** [comment]
> You code doesn't compile as `CloudLoginDTO` uses the wrong `NotNull` import:
> `import org.jetbrains.annotations.NotNull;`

This was changed some time ago.. should no longer be the case

**Review by fwolter — APPROVED** [review]
LGTM

There are some checkstyle warnings in the binding, which can be easily fixed. You could take a look at `target/code-analysis/report.html`.

**Pull request merged**